src/braket/tracking/tracker.py

```diff
@@ -0,0 +1,276 @@
+        """
+        total_cost = Decimal(0)
+        for task_arn, details in self._resources.items():
+            if "qpu" in details["device"]:
+                total_cost = total_cost + _get_qpu_task_cost(task_arn, details)
+        return total_cost
+
+    def simulator_tasks_cost(self) -> Decimal:
+        """
+        Estimate cost of all quantum tasks tracked by this tracker using Braket simulator devices.
+
+        Returns:
+            Decimal: The estimated total cost in USD
+        """
+        total_cost = Decimal(0)
+        for task_arn, details in self._resources.items():
+            if "qpu" not in details["device"]:
+                total_cost = total_cost + _get_simulator_task_cost(task_arn, details)
+        return total_cost
+
+    def quantum_tasks_statistics(self) -> Dict[str, Dict[str, Any]]:
+        """
+        Get a summary of quantum tasks grouped by device.
+
+        Returns:
+            Dict[str,Dict[str,Any]] : A dictionary where each key is a device arn, and maps to
+                a dictionary sumarizing the tasks run on the device. The summary includes the
+                total shots sent to the device and the most recent status of the quantum tasks
+                created on this device. For finished tasks on simulator devices, the summary
+                also includes the duration of the simulation in seconds.
+
+        Example:
+            >>> tracker.quantum_tasks_statistics()
+            {'qpu_device_foo':
+                {'shots' : 1000,
+                 'tasks' : { 'COMPLETED' : 4,
+                             'QUEUED' : 1 },
+                },
+             'simulator_device_bar':
+                {'shots' : 1000
+                 'tasks' : { 'COMPLETED' : 2,
+                              'CREATED' : 1},
+                 'execution_duration' : 5.432,
+                 'billed_execution_duration' : 6.543}}
+        """
+        stats = {}
+        for _, details in self._resources.items():
+
+            device_stats = stats.get(details["device"], {})
+
+            shots = device_stats.get("shots", 0) + details["shots"]
+            device_stats["shots"] = shots
+
+            task_states = device_stats.get("tasks", {})
+            task_states[details["status"]] = task_states.get(details["status"], 0) + 1
+            device_stats["tasks"] = task_states
+
+            if "execution_duration" in details:
+                duration = device_stats.get("execution_duration", 0) + details["execution_duration"]
+                billed_duration = (
+                    device_stats.get("billed_execution_duration", 0) + details["billed_duration"]
+                )
+
+                device_stats["execution_duration"] = duration / 1000
+                device_stats["billed_execution_duration"] = billed_duration / 1000
+
+            stats.setdefault(details["device"], {}).update(device_stats)
+
+        return stats
+
+
+def _get_qpu_task_cost(task_arn: str, details: dict) -> Decimal:
```

Comment:
What if the task has been cancelled / fails ?

Reply:
Added a check for cancelled/failed state.